tests/unit/admin/tasks/test-class-wc-payments-task-disputes.php

```diff
@@ -30,67 +30,207 @@ public function tear_down() {
-			->willReturn( [] );
+	public function test_disputes_task_with_single_dispute_within_7days() {
+		$disputes_task        = new WC_Payments_Task_Disputes();
+		$mock_active_disputes = [
+			[
+				'wcpay_disputes_cache_id' => 21,
+				'stripe_account_id'       => 'acct_abc',
+				'dispute_id'              => 'dp_2',
+				'charge_id'               => 'ch_2',
+				'amount'                  => 2000,
+				'currency'                => 'eur',
+				'reason'                  => 'product_not_received',
+				'source'                  => 'visa',
+				'order_number'            => 14,
+				'customer_name'           => 'customer',
+				'customer_email'          => 'email@email.com',
+				'customer_country'        => 'US',
+				'status'                  => 'needs_response',
+				'created'                 => gmdate( 'Y-m-d H:i:s', strtotime( '-14 days' ) ),
+				'due_by'                  => gmdate( 'Y-m-d H:i:s', strtotime( '+6 days' ) ),
+			],
+		];
+		$this->mock_cache->method( 'get_or_add' )->willReturn(
+			$mock_active_disputes
+		);
+
+		$this->assertEquals( 'Respond to a dispute for 20,00 €', $disputes_task->get_title() );
+		// "By <date> – <count> days left to respond"
+		$this->assertMatchesRegularExpression( '/By \w{3} \d{1,2}, \d{4} – \d+ days left to respond/', $disputes_task->get_additional_info() );
+		$this->assertEquals( true, $disputes_task->can_view() );
 
-		$this->assertEquals( '0 disputed payments need your response', $disputes_task->get_title() );
-		$this->assertEquals( false, $disputes_task->can_view() );
 	}
 
-	public function test_disputes_task_returns_the_correct_content_with_single_item_on_needs_response() {
-		$disputes_task = new WC_Payments_Task_Disputes();
+	public function test_disputes_task_with_single_dispute_within_24h() {
+		$disputes_task        = new WC_Payments_Task_Disputes();
+		$mock_active_disputes = [
+			[
+				'wcpay_disputes_cache_id' => 21,
+				'stripe_account_id'       => 'acct_abc',
+				'dispute_id'              => 'dp_2',
+				'charge_id'               => 'ch_2',
+				'amount'                  => 2000,
+				'currency'                => 'eur',
+				'reason'                  => 'product_not_received',
+				'source'                  => 'visa',
+				'order_number'            => 14,
+				'customer_name'           => 'customer',
+				'customer_email'          => 'email@email.com',
+				'customer_country'        => 'US',
+				'status'                  => 'needs_response',
+				'created'                 => gmdate( 'Y-m-d H:i:s', strtotime( '-14 days' ) ),
+				'due_by'                  => gmdate( 'Y-m-d H:i:s', strtotime( '+23 hours' ) ),
+			],
+		];
+		$this->mock_cache->method( 'get_or_add' )->willReturn(
+			$mock_active_disputes
+		);
+
+		$this->assertEquals( 'Respond to a dispute for 20,00 € – Last day', $disputes_task->get_title() );
+		// "Respond today by <time> <AM|PM>"
+		$this->assertMatchesRegularExpression( '/Respond today by \d{1,2}:\d{2} (AM|PM)/', $disputes_task->get_additional_info() );
```

Comment:
Would be nice if there was an easy way to mock the datetime in PHP, without third party libraries.